enhancements/installer/packet-ipi.md

```diff
@@ -0,0 +1,294 @@
+
+We can help these users simplify the process of installing OpenShift on Packet
+by introducing installer-provisioned options that take advantage of various
+models of compute, memory, GPU, or storage classes of devices, hardware and IP
+reservations, virtual networks, and fast direct-attached storage.
+
+Currently, users who want to [deploy OpenShift on
+Packet](https://www.packet.com/solutions/openshift/) must follow the [OpenShift
+via Terraform on
+Packet][https://github.com/RedHatSI/openshift-packet-deploy/blob/master/terraform/README.md]
+UPI instructions. This process is not streamlined and can not be integrated into
+a simple experience like the [Try OpenShift 4](https://www.openshift.com/try)
+workflow.
+
+### Goals
+
+The main goal of the Packet IPI is to provide users with an easier path to
+running Red Hat OpenShift 4 on a dynamically provisioned, horizontally scalable,
+bare metal architecture in [Packet data
+centers](https://www.packet.com/developers/docs/getting-started/datacenters/).
+
+As a first step, we would add the Packet IPI to the installer codebase. With
+accompanying documentation, this makes a the Packet option available through the
+CLI. This would enable users to deploy OpenShift with IPI in a way that is very
+similar to UPI but simplifies the process.
+
+Following other IPI installers, this first step would include
+
+- Making Packet IPI documentation available here:
+  <https://github.com/openshift/installer/blob/master/docs/user/packet/install_ipi.md>
+- Adapt Packet's sig-lifecyclce ClusterAPI v1alpha2 for use as an OpenShift's
+  ClusterAPI v1beta1 Machine driver
+- Prepare the Terraform code and Packet types necessary for an IPI installer
+- Making a CI job executing the provisioning scripts to test the IPI installer
+
+### Non-Goals
+
+It is outside the scope of this enhancement to provide details about the
+installation and infrastructure elements that have been highlighted in [Open
+Questions](#open-questions) and [Implementation
+Details/Notes/Constraints](#implementation-details-notes-constraints), even for
+those elements that are required (e.g. DNS).
+
+When open questions are explored and resolved, they will be added to this
+enhancement request or they will be surfaced in additional enhancement requests,
+as needed.
+
+## Proposal
+
+- Define and implement Packet types in the openshift/installer
+  - High level variables should include:
+    - [API Key](https://www.packet.com/developers/docs/API/) (required)
+    - [Project
+      ID](https://www.packet.com/developers/docs/getting-started/support/organizations-and-users/)
+      (required) (Options to create a project could be presented)
+  - Boostrap node variables include:
+    - Device plan (required, defaulting to minimum required)
+    - Facility (required)
+    - Virtual Network (optional, defaulted to a new virtual network)
+  - Control plane variables rely on boostrap variables and add:
```

Comment:
Does "rely on bootstrap variables" mean that it will inherit the same settings, enforcing common settings for the whole cluster?  Or is this a shorthand way of saying "all of that stuff plus some more" ?

Reply:
I was thinking inheritance but I left my wording open in case that is not an option or not the best choice.